src/other/Opportunity.js

```diff
@@ -0,0 +1,381 @@
+"use strict";
+(function(root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/SVGWidget", "../common/Utility", "../common/PropertyExt", "css!./Opportunity"], factory);
+    } else {
+        root.graph_Opportunity = factory(root.d3, root.common_SVGWidget, root.common_Utility, root.common_PropertyExt);
+    }
+}(this, function(d3, SVGWidget, Utility, PropertyExt) {
+var groupCount = 7;
+    function Column(owner) {
+        PropertyExt.call(this);
+        this._owner = owner;
+    }
+    Column.prototype = Object.create(PropertyExt.prototype);
+    Column.prototype.constructor = Column;
+    Column.prototype._class += " graph_Opportunity.Column";
+    Column.prototype.publish("headerLabel", null, "string", "Header value of a table", function() { return this._owner ? this._owner.columns() : [];}, { tags: ["Basic"], optional: true });
+    function mouseHoverColumn(owner) {
+        PropertyExt.call(this);
+        this._owner = owner;
+    }
+    mouseHoverColumn.prototype = Object.create(PropertyExt.prototype);
+    mouseHoverColumn.prototype.constructor = mouseHoverColumn;
+    mouseHoverColumn.prototype._class += " graph_Opportunity.mouseHoverColumn";
+    mouseHoverColumn.prototype.publish("hoverValue", null, "string", "Hover value of a table", function() { return this._owner ? this._owner.columns() : []; }, { tags: ["Basic"], optional: true });
+    mouseHoverColumn.prototype.publish("hoverList", null, "set", "Hover value of a table", function() { return this._owner ? this._owner.getIds() : []; }, { tags: ["Basic"], optional: true });
+    function columnDropdown(owner) {
+        PropertyExt.call(this);
+        this._owner = owner;
+    }
+    columnDropdown.prototype = Object.create(PropertyExt.prototype);
+    columnDropdown.prototype.constructor = columnDropdown;
+    columnDropdown.prototype._class += " graph_Opportunity.columnDropdown";
+    columnDropdown.prototype.publish("columnIndex", null, "number", "Column index for display context data based on column dropdown list selction",{}, {tags: ["Basic", "Shared"], });
+    columnDropdown.prototype.publish("columnDropdownList", null, "set", "column value of a table",function() { return this._owner ? this._owner.getIds() : []; },{tags: ["Basic"], optional: true,});
+
+    function Opportunity(target) {
+        SVGWidget.call(this);
+    }
+    Opportunity.prototype = Object.create(SVGWidget.prototype);
+    Opportunity.prototype.constructor = Opportunity;
+    Opportunity.prototype.Column = Column;
+    Opportunity.prototype.mouseHoverColumn = mouseHoverColumn;
+    Opportunity.prototype.columnDropdown = columnDropdown;
+    Opportunity.prototype._class += " graph_Opportunity";
+    Opportunity.prototype.publish("id", "id", "set", "Id for label in Opportunity", function() { return this.getIds(); }, { tags: ["Basic", "Shared"]});
```

Comment:
"id" is probably a bad choice for a published property as all widgets already have an "id" property.,